packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWTable/CWTable.tsx

```diff
@@ -0,0 +1,251 @@
+that is not a string, number, or date, sorting won't actually work (i.e. items will be re-ordered, but in a mysterious
+way), even if you set the `sortable` attribute to true. However, the arrow will be there, which makes for a terrble
+user experience. Be certain that the column's data can be sorted before designating it as sortable.
+
+The rowData array has no required keys on the objects within it. However, if any of your columns have data that is
+displayed along with an avatar, you'll need to make use of the `avatars` object. `avatars` is an object with keys
+that correspond to keys in the data that have avatars. For instance, if you would like to display the logo of
+Stark Industries next to the name 'Stark Industries', insert a key-value pair into the `avatars` object where the
+key matches the associated key in the data and the value is the URL where the avatar can be found (see above
+data structure).
+*/
+
+import React, { useMemo, useState } from 'react';
+import { CWIcon } from '../../cw_icons/cw_icon';
+import './CWTable.scss';
+import { Avatar } from '../../../Avatar';
+import { ComponentType } from '../../types';
+import clsx from 'clsx';
+
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table';
+
+type ColumnDescriptor = {
+  key: string;
+  header: string;
+  numeric: boolean;
+  sortable: boolean;
+};
+
+type RowData = {
+  avatars?: object;
+};
+
+type TableProps = {
+  columnInfo: ColumnDescriptor[];
+  rowData: any[];
+};
+
+export const CWTable = ({ columnInfo, rowData }: TableProps) => {
+  const [sorting, setSorting] = useState<SortingState>([
+    {
+      id: columnInfo[0].key,
+      desc: false,
+    },
+  ]);
+
+  const columns = useMemo<ColumnDef<unknown, any>[]>(
+    () =>
+      columnInfo.map((col) => {
+        return {
+          accessorKey: col.key,
+          header: col.header,
+          cell: (info) => {
+            const currentRow = info.row.original as RowData;
+            const avatarUrl = currentRow.avatars?.[col.key];
+
+            if (col.numeric) {
+              return <div className="numeric">{info.getValue()}</div>;
+            } else if (avatarUrl) {
+              return (
+                <div className="avatar-cell">
+                  <Avatar url={avatarUrl} size={20} />
+                  <div className="text">{info.getValue()}</div>
+                </div>
+              );
+            } else {
+              return info.getValue();
+            }
+          },
+          footer: (props) => props.column.id,
+          enableSorting: col.sortable,
+        };
+      }),
+    []
+  );
+
+  const table = useReactTable({
+    data: rowData,
+    columns,
+    state: {
+      sorting,
+    },
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    debugTable: true,
```

Comment:
Maybe we should set debugTable: true only for dev mode. 

I was getting these in the console

<img width="471" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/b29086a3-e616-4101-9065-98ef1923fbc9">

Reply:
agree

Reply:
Hey @mzparacha @masvelio - `debugTable` only works in dev mode (if you take a look on the demo branch, you shouldn't see any console messages).